src/exec/noexec.rs

```diff
@@ -170,7 +170,7 @@ union SingleRightAnciliaryData {
 }
 
 /// Receives a raw file descriptor from the provided UnixStream
-fn receive_fd(rx_fd: UnixStream) -> RawFd {
+fn receive_fd(rx_fd: UnixStream) -> io::Result<RawFd> {
```

Comment:
I don't think this adds much. Either way the error is fatal.

Reply:
I just found the explicit panic surprising at that point in the file, the `.expect()` is less surprising given that it happens inside a spawned thread. 

Should we actually abort there, instead of panic/.expect?

Reply:
The reason I was using abort everywhere is because of the prototyping I did with running it as part of an LD_PRELOAD loaded shared library. I tried to keep it as small as possible by avoiding the entire panic machinery.

Reply:
I think we should panic everywhere instead.

Reply:
The panic in the thread will simply hang the process, which (**if** this would ever happen) is quite confusing for a user; an `abort` finishes it. Similarly for the `unreachable!` in fact.

Reply:
Note: I've removed this particular commit from this list of suggestions. We can merge the outstanding PR and worry about the current `panics` and `unwraps` later.

Reply:
Why would a panic cause the process to hang? It should abort the process after printing a panic message.